test/performance/perf-upsert/src/main.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Performance test for Materialize Upsert sources
+
+#![deny(missing_debug_implementations, missing_docs)]
+
+use std::process;
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::Result;
+use rand::Rng;
+use structopt::StructOpt;
+
+use test_util::kafka::kafka_client;
+
+use crate::config::{Args, KafkaConfig, MzConfig};
+use crate::mz_client::MzClient;
+
+mod config;
+mod mz_client;
+
+#[tokio::main]
+async fn main() {
+    if let Err(e) = run().await {
+        println!("ERROR: {}", e);
```

Comment:
nit TIOLI: human logs should always go to stderr, it makes it easier to add machine-focused output later, and prevents stdout and stderr logs (which are what are emitted by logging infrastructure by default) from getting interleaved in surprising orders:  `eprintln!`